common/web/types/src/util/compiler-interfaces.ts

```diff
@@ -6,6 +6,27 @@ export interface CompilerEvent {
   message: string;
 };
 
+export enum CompilerErrorSeverity {
+  Info =          0x000000,
+  Hint =          0x100000,
+  Warn =          0x200000,
+  Error =         0x300000,
+  Fatal =         0x400000,
+  Severity_Mask = 0xF00000, // includes reserved bits
+  Error_Mask =    0x0FFFFF,
+};
+
+export enum CompilerErrorNamespace {
+  /**
+   * KMC errors between 0x0000…0x0FFF
+   */
+  Kmc = 0x0000,
```

Comment:
This should probably be `KeyboardCompiler` or perhaps (I'd prefer not) `KmcKeyboard`, not `Kmc`.

```suggestion
  /**
   * kmc-keyboard errors between 0x0000…0x0FFF
   */
  KeyboardCompiler = 0x0000,
```